### ISSUE TITLE
Uniquify identical completions

### ISSUE THREAD

**Issue by meedstrom** [meedstrom/org-node#62]
How? There's the concept of a "tofu", an invisible character added to each completion candidate, to be stripped later. Prior art in Consult.

**Comment by semilin** [comment]
Why not include a unique property in identically named candidates? The `CREATED` property would make most sense to me, but you could also show the `ID`. Either way, I think distinguishing these candidates somehow would be better than showing them as identical.

**Comment by meedstrom** [comment]
True, good to distinguish them.  An annotation-function (like `org-node-affixation-fn`, just with shorter annotations) could also suffice to distinguish them.

Either way, I guess the hard part will be stripping the tofu or unique property, prior to returning the candidate. Haven't learned to do that.

**Comment by fleimgruber** [comment]
For inspiration (`org-ql` was also inspiration for `org-roam-ql` mentioned in https://github.com/meedstrom/org-node/issues/87) maybe look at: https://github.com/alphapapa/org-ql/blob/98c62ab0a6c084ae4132110e24d9fe1ace91d363/org-ql-completing-read.el#L191?

**Comment by krisbalintona** [comment]
Question: why are completion candidates node titles instead of the nodes themselves? For instance, `org-roam-node-read` has org-roam nodes as completing-read candidates.

**Comment by meedstrom** [comment]
You'd think so, it's just confusingly written.  It's passed thru `org-roam-node-read--to-candidate`, which returns strings.  Anyway, it has to be strings.  If you pass e.g. a hash table to completing-read, it will actually just use the hash table keys.
